source/jormungandr/tests/departure_board_tests.py

```diff
@@ -1104,3 +1103,559 @@ def test_routes_schedule_with_invalid_forbidden_uri(self):
+        assert stop_schedules[0]['last_datetime']['date_time'] == "20170105T062500"
+        # since it's the last stop_point, we cannot get on the bus....
+        assert 'first_datetime' not in stop_schedules[1]
+        assert 'last_datetime' not in stop_schedules[1]
+
+    def test_first_last_datetime_another_validity(self):
+        if not app.config['USE_SERPY']:
+            return
+
+        """              20170105
+       from_datetime 20170105T0730       
+                  |  
+                  v
+        (Open)                        (Close)
+        06:00                         00:55  
+         X_S1 ------------------------ X_S4
+
+        """
+        from_datetime = "20170105T073000"
+        response = self.query_region("stop_points/X_S2/stop_schedules?from_datetime=20170105T0730"
+                                     "&data_freshness=base_schedule".format(from_datetime))
+
+        stop_schedules = response['stop_schedules']
+        assert len(stop_schedules) == 2
+        assert stop_schedules[0]['first_datetime']['date_time'] == "20170106T063500"
+        assert stop_schedules[0]['last_datetime']['date_time'] == "20170106T065500"
+        assert stop_schedules[1]['first_datetime']['date_time'] == "20170105T234000"
+        assert stop_schedules[1]['last_datetime']['date_time'] == "20170105T234500"
+
+        response = self.query_region("stop_points/X_S1/stop_schedules?from_datetime={}"
+                                     "&data_freshness=base_schedule".format(from_datetime))
+
+        stop_schedules = response['stop_schedules']
+        assert len(stop_schedules) == 2
+        assert stop_schedules[0]['first_datetime']['date_time'] == "20170106T060500"
+        assert stop_schedules[0]['last_datetime']['date_time'] == "20170106T062500"
+        # since it's the last stop_point, we cannot get on the bus....
+        assert 'first_datetime' not in stop_schedules[1]
+        assert 'last_datetime' not in stop_schedules[1]
+
+    def test_first_last_datetime_just_after_midnight(self):
+        if not app.config['USE_SERPY']:
+            return
+
+        """              20170101 (the beginning of production)
+                           from_datetime 20170102T000001
+                                      |  
+                                      v
+        (Open)                        (Close)
+        06:00                         00:55  
+         X_S1 ------------------------ X_S4
+
+        """
+        from_datetime = "20170102T000001"
+        response = self.query_region("stop_points/X_S3/stop_schedules?from_datetime={}"
+                                     "&data_freshness=base_schedule".format(from_datetime))
+
+        stop_schedules = response['stop_schedules']
+        assert stop_schedules
+        assert stop_schedules[0]['first_datetime']['date_time'] == "20170102T235000"
+        assert stop_schedules[0]['last_datetime']['date_time'] == "20170103T001000"
```

Comment:
You could test on stop_schedules[1] also, no? You would have 20170102T064000 - 20170102T064500 I guess (same below).

Reply:
done